src/main/cpp/util/md5.h

```diff
@@ -24,6 +24,22 @@
 #define __alignof__ __alignof
 #endif  // _MSC_VER && !__alignof__
 
+#if defined(__linux__)
+#include <endian.h>
+#elif defined(__FreeBSD__) || defined(__OpenBSD__)
+#include <sys/endian.h>
+#elif defined(__APPLE__) || defined(_WIN32)
+// Hopefully OSX and Windows will keep running solely on little endian CPUs, so:
+#define le16toh(x) (x)
+#define le32toh(x) (x)
+#define le64toh(x) (x)
+#define htole16(x) (x)
+#define htole32(x) (x)
```

Comment:
Do you need anything except this?

Reply:
No, we only need this one. I've removed other unused `#define` directives. Thanks for pointing out!

Reply:
How are you computing the endian-ness. 
AFAIK Windows is always little endian today, but Apple does have a good header for it.
We might want https://opensource.apple.com/source/CarbonHeaders/CarbonHeaders-18.1/Endian.h